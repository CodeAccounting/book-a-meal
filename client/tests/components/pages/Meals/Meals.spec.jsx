import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import MealsComponent from '../../../../src/components/pages/Meals';
import MealsContainer from '../../../../src/containers/pages/Meals';
import { caterer, customer, mealsObj, initialState, metadata } from '../../../setup/mockData';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  ...initialState, meals: { ...initialState.meals, items: mealsObj.meals, metadata }
});
const { now } = Date;

describe('Meals', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly when not fetching', () => {
    const toggleMock = jest.fn();

    const shallowWrapper = shallow(<MealsComponent
      user={caterer}
      logout={jest.fn()}
      fetchMeals={jest.fn()}
      {...mealsObj}
      isFetching={false}
      submitting={false}
      uploading={false}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('MealCard')).toBeTruthy();
  });

  it('does not call fetchMeals on component mount when metadata is not available', () => {
    const toggleMock = jest.fn();
    const fetchMealsMock = jest.fn();

    shallow(<MealsComponent
      user={caterer}
      logout={jest.fn()}
      fetchMeals={fetchMealsMock}
      {...mealsObj}
      isFetching={false}
      submitting={false}
      uploading={false}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    expect(fetchMealsMock).not.toHaveBeenCalled();
  });

  it('calls fetchMeals on component mount when metadata is available', () => {
    const toggleMock = jest.fn();
    const fetchMealsMock = jest.fn();

    shallow(<MealsComponent
      user={caterer}
      logout={jest.fn()}
      fetchMeals={fetchMealsMock}
      {...mealsObj}
      isFetching={false}
      submitting={false}
      uploading={false}
      toggleModal={toggleMock}
      metadata={{}}
    />);

    expect(fetchMealsMock).toHaveBeenCalled();
  });

  it('calls fetchMeals when loadMoreMeals is called', () => {
    const toggleMock = jest.fn();
    const fetchMealsMock = jest.fn();

    const shallowWrapper = shallow(<MealsComponent
      user={caterer}
      logout={jest.fn()}
      fetchMeals={fetchMealsMock}
      {...mealsObj}
      isFetching={false}
      submitting={false}
      uploading={false}
      toggleModal={toggleMock}
      metadata={{}}
    />);

    shallowWrapper.instance().loadMoreMeals();

    expect(fetchMealsMock).toHaveBeenCalledTimes(2);
  });

  it('calls toggleModal when button is clicked', () => {
    const toggleMock = jest.fn();
    const wrapper = shallow(<MealsComponent
      user={caterer}
      logout={jest.fn()}
      fetchMeals={jest.fn()}
      {...mealsObj}
      isFetching={false}
      submitting={false}
      uploading={false}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    wrapper.find('#add-meal-btn').simulate('click');
    expect(toggleMock).toHaveBeenCalled();
  });

  it('calls toggleModal when edit meal button is clicked', () => {
    const comp = (
      <Provider store={store}>
        <MealsContainer
          user={caterer}
          logout={jest.fn()}
          fetchMeals={jest.fn()}
          {...mealsObj}
          isFetching={false}
        />
      </Provider>
    );
    const wrapper = mount(comp, rrcMock.get());

    const toggleSpy = jest.spyOn(wrapper.find(MealsComponent).instance(), 'toggleModal');

    wrapper.find('#edit-meal').at(0).simulate('click');
    expect(toggleSpy).toHaveBeenCalled();
  });

  it('redirects User to homepage if user role is customer', () => {
    const toggleMock = jest.fn();
    const shallowWrapper = shallow(<MealsComponent
      user={customer}
      logout={jest.fn()}
      fetchMeals={jest.fn()}
      {...mealsObj}
      isFetching
      submitting={false}
      uploading={false}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    expect(shallowWrapper.find('Redirect')).toBeTruthy();
  });

  it('renders Preloader when fetching', () => {
    const toggleMock = jest.fn();
    const shallowWrapper = shallow(<MealsComponent
      user={caterer}
      logout={jest.fn()}
      fetchMeals={jest.fn()}
      {...mealsObj}
      isFetching
      submitting={false}
      uploading={false}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('Preloader')).toBeTruthy();
  });

  it('renders message when not fetching and there are no meals', () => {
    const toggleMock = jest.fn();
    const shallowWrapper = shallow(<MealsComponent
      user={caterer}
      logout={jest.fn()}
      fetchMeals={jest.fn()}
      meals={[]}
      isFetching
      submitting={false}
      uploading={false}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('p').text()).toEqual('You Do Not Have Any Meals Yet');
  });

  it('renders connected component', () => {
    const dispatchMock = jest.fn();
    const comp = (
      <Provider store={store}>
        <MealsContainer
          user={caterer}
          dispatch={dispatchMock}
          {...mealsObj}
          isFetching={false}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
