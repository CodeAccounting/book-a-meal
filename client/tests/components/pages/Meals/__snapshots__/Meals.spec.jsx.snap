// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meals renders Preloader when fetching 1`] = `
<React.Fragment>
  <CatererView
    isFetching={true}
    logout={[MockFunction]}
    showTime={true}
    type="meals"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <React.Fragment>
      <div
        className="content-wrapper meals caterer-meals"
      >
        <div
          className="top"
        >
          <button
            className="btn btn-pri"
            id="add-meal-btn"
            onClick={[Function]}
          >
            Add a New Meal
          </button>
        </div>
        <div
          className="card-group meals-wrapper"
          id="card-group"
        >
          <MealCard
            key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
            meal={
              Object {
                "description": "Sharwama contains no animal products, perfect for dieters",
                "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                "price": "1200.00",
                "title": "Vegetable Sharwama and Guava Smoothie",
                "vegetarian": true,
              }
            }
            toggleModal={[Function]}
            type="caterer"
          />
          <MealCard
            key="81211c24-51c0-46ec-b1e0-18db55880958"
            meal={
              Object {
                "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                "price": "1500.00",
                "title": "Jollof Rice, Beef and Plantain",
                "vegetarian": false,
              }
            }
            toggleModal={[Function]}
            type="caterer"
          />
        </div>
      </div>
    </React.Fragment>
  </CatererView>
  <Connect(Modal)
    submitError={null}
    submitting={false}
  />
</React.Fragment>
`;

exports[`Meals renders connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Meals)
    dispatch={[MockFunction]}
    isFetching={false}
    meals={
      Array [
        Object {
          "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
          "id": "81211c24-51c0-46ec-b1e0-18db55880958",
          "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
          "price": "1500.00",
          "title": "Jollof Rice, Beef and Plantain",
          "vegetarian": false,
        },
        Object {
          "description": "Sharwama contains no animal products, perfect for dieters",
          "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
          "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
          "price": "1200.00",
          "title": "Vegetable Sharwama and Guava Smoothie",
          "vegetarian": true,
        },
      ]
    }
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <Meals
      dispatch={[MockFunction]}
      fetchMeals={[Function]}
      isFetching={false}
      logout={[Function]}
      meals={
        Array [
          Object {
            "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
            "id": "81211c24-51c0-46ec-b1e0-18db55880958",
            "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
            "price": "1500.00",
            "title": "Jollof Rice, Beef and Plantain",
            "vegetarian": false,
          },
          Object {
            "description": "Sharwama contains no animal products, perfect for dieters",
            "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
            "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
            "price": "1200.00",
            "title": "Vegetable Sharwama and Guava Smoothie",
            "vegetarian": true,
          },
        ]
      }
      submitError={null}
      submitting={false}
      toggleModal={[Function]}
      user={
        Object {
          "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
          "businessName": "Food Circle",
          "businessPhoneNo": "+2348166557788",
          "email": "food@circle.com",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "role": "caterer",
        }
      }
    >
      <CatererView
        isFetching={false}
        logout={[Function]}
        showTime={true}
        type="meals"
        user={
          Object {
            "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
            "businessName": "Food Circle",
            "businessPhoneNo": "+2348166557788",
            "email": "food@circle.com",
            "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
            "role": "caterer",
          }
        }
      >
        <div
          className="admin"
        >
          <Header
            dateType="meals"
            showTime={true}
            type="caterer"
          >
            <div
              className="header"
            >
              <div
                className="navbar"
              >
                <Connect(CatererHeader)
                  dateType="meals"
                  showTime={true}
                >
                  <CatererHeader
                    currentDay="1970-01-01"
                    dateType="meals"
                    dispatch={[Function]}
                    showTime={true}
                  >
                    <div
                      className="menu-control d-none-md"
                    >
                      <a
                        className="d-none-md"
                        href="#sidenav"
                      >
                        <img
                          alt="toggler"
                          src="./img/toggler.png"
                        />
                      </a>
                    </div>
                    <div
                      className="page-title d-flex-md"
                    >
                      <div
                        className="date"
                      >
                        <h2>
                          Thursday, 1st January 1970
                              
                          00:00
                        </h2>
                      </div>
                    </div>
                  </CatererHeader>
                </Connect(CatererHeader)>
              </div>
            </div>
          </Header>
          <div
            className="content"
          >
            <SideNav
              active="meals"
              logout={[Function]}
              user={
                Object {
                  "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "businessPhoneNo": "+2348166557788",
                  "email": "food@circle.com",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "role": "caterer",
                }
              }
            >
              <div
                className="sidenav"
                id="sidenav"
              >
                <div
                  className="sidenav-content"
                >
                  <div
                    className="sidenav-header"
                  >
                    <div
                      className="close d-none-md"
                    >
                      <a
                        href="#"
                      >
                        ×
                      </a>
                       
                    </div>
                    <div
                      className="sidenav-title"
                    >
                      <h3>
                        <a
                          href="/"
                        >
                          BOOK-A-MEAL
                        </a>
                      </h3>
                      <div
                        className="username-circle"
                      >
                        <p>
                          F
                        </p>
                      </div>
                      <p>
                        Food Circle
                      </p>
                      <span>
                        <Link
                          className={null}
                          clickHandler={[Function]}
                          id={null}
                        >
                          <button
                            className="link-btn null"
                            id={null}
                            onClick={[Function]}
                          >
                            Logout
                          </button>
                        </Link>
                      </span>
                    </div>
                  </div>
                  <div
                    className="sidenav-body"
                  >
                    <a
                      className="menu-item false"
                      href="/"
                    >
                      Dashboard
                    </a>
                    <a
                      className="menu-item active"
                      href="/meals"
                    >
                      Meals
                    </a>
                    <a
                      className="menu-item false"
                      href="/menu"
                    >
                      Menu
                    </a>
                    <a
                      className="menu-item false"
                      href="/orders"
                    >
                      Order History
                    </a>
                  </div>
                </div>
              </div>
            </SideNav>
            <div
              className="content-wrapper"
              id="has-modal"
            >
              <div
                className="content-wrapper meals caterer-meals"
              >
                <div
                  className="top"
                >
                  <button
                    className="btn btn-pri"
                    id="add-meal-btn"
                    onClick={[Function]}
                  >
                    Add a New Meal
                  </button>
                </div>
                <div
                  className="card-group meals-wrapper"
                  id="card-group"
                >
                  <MealCard
                    key="81211c24-51c0-46ec-b1e0-18db55880958"
                    meal={
                      Object {
                        "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                        "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                        "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                        "price": "1500.00",
                        "title": "Jollof Rice, Beef and Plantain",
                        "vegetarian": false,
                      }
                    }
                    toggleModal={[Function]}
                    type="caterer"
                  >
                    <div
                      className="meal-card"
                      id="meal-card"
                    >
                      <div
                        className="meal-card-header"
                      >
                        <img
                          alt="meal"
                          src="http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg"
                        />
                        <Dropdown
                          content={
                            <UNDEFINED>
                              <Link
                                className={null}
                                clickHandler={[Function]}
                                href="#add-edit-modal"
                                id="edit-meal"
                              >
                                Edit
                              </Link>
                              <Link
                                className={null}
                                clickHandler={[Function]}
                                href="#add-edit-modal"
                                id="delete-meal"
                              >
                                Delete
                              </Link>
                            </UNDEFINED>
                          }
                          toggler={
                            <UNDEFINED>
                              …
                            </UNDEFINED>
                          }
                          type="card"
                        >
                          <div
                            className="dropdown card-dropdown"
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          >
                            <Link
                              className="link"
                              clickHandler={[Function]}
                              id="dropdown-toggler"
                            >
                              <button
                                className="link-btn link"
                                id="dropdown-toggler"
                                onClick={[Function]}
                              >
                                …
                              </button>
                            </Link>
                            <div
                              className="dropdown-content meal"
                              id="dropdown-content"
                            >
                              <Link
                                className={null}
                                clickHandler={[Function]}
                                href="#add-edit-modal"
                                id="edit-meal"
                              >
                                <button
                                  className="link-btn null"
                                  id="edit-meal"
                                  onClick={[Function]}
                                >
                                  Edit
                                </button>
                              </Link>
                              <Link
                                className={null}
                                clickHandler={[Function]}
                                href="#add-edit-modal"
                                id="delete-meal"
                              >
                                <button
                                  className="link-btn null"
                                  id="delete-meal"
                                  onClick={[Function]}
                                >
                                  Delete
                                </button>
                              </Link>
                            </div>
                          </div>
                        </Dropdown>
                        <div
                          className="menu-card-title"
                        >
                          <p>
                            Jollof Rice, Beef and Plantain
                          </p>
                        </div>
                      </div>
                      <div
                        className="meal-card-body"
                      >
                        <div>
                          <h3>
                            ₦ 
                            1500.00
                          </h3>
                          <p>
                            Jollof Rice, Beef and Plantain. 2 pieces of beef per plate
                          </p>
                        </div>
                      </div>
                    </div>
                  </MealCard>
                  <MealCard
                    key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
                    meal={
                      Object {
                        "description": "Sharwama contains no animal products, perfect for dieters",
                        "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                        "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                        "price": "1200.00",
                        "title": "Vegetable Sharwama and Guava Smoothie",
                        "vegetarian": true,
                      }
                    }
                    toggleModal={[Function]}
                    type="caterer"
                  >
                    <div
                      className="meal-card"
                      id="meal-card"
                    >
                      <div
                        className="meal-card-header"
                      >
                        <img
                          alt="meal"
                          src="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350"
                        />
                        <Dropdown
                          content={
                            <UNDEFINED>
                              <Link
                                className={null}
                                clickHandler={[Function]}
                                href="#add-edit-modal"
                                id="edit-meal"
                              >
                                Edit
                              </Link>
                              <Link
                                className={null}
                                clickHandler={[Function]}
                                href="#add-edit-modal"
                                id="delete-meal"
                              >
                                Delete
                              </Link>
                            </UNDEFINED>
                          }
                          toggler={
                            <UNDEFINED>
                              …
                            </UNDEFINED>
                          }
                          type="card"
                        >
                          <div
                            className="dropdown card-dropdown"
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          >
                            <Link
                              className="link"
                              clickHandler={[Function]}
                              id="dropdown-toggler"
                            >
                              <button
                                className="link-btn link"
                                id="dropdown-toggler"
                                onClick={[Function]}
                              >
                                …
                              </button>
                            </Link>
                            <div
                              className="dropdown-content meal"
                              id="dropdown-content"
                            >
                              <Link
                                className={null}
                                clickHandler={[Function]}
                                href="#add-edit-modal"
                                id="edit-meal"
                              >
                                <button
                                  className="link-btn null"
                                  id="edit-meal"
                                  onClick={[Function]}
                                >
                                  Edit
                                </button>
                              </Link>
                              <Link
                                className={null}
                                clickHandler={[Function]}
                                href="#add-edit-modal"
                                id="delete-meal"
                              >
                                <button
                                  className="link-btn null"
                                  id="delete-meal"
                                  onClick={[Function]}
                                >
                                  Delete
                                </button>
                              </Link>
                            </div>
                          </div>
                        </Dropdown>
                        <div
                          className="menu-card-title"
                        >
                          <p>
                            Vegetable Sharwama and Guava Smoothie
                          </p>
                        </div>
                      </div>
                      <div
                        className="meal-card-body"
                      >
                        <div>
                          <h3>
                            ₦ 
                            1200.00
                          </h3>
                          <p>
                            Sharwama contains no animal products, perfect for dieters
                          </p>
                        </div>
                      </div>
                    </div>
                  </MealCard>
                </div>
              </div>
            </div>
          </div>
          <Footer>
            <div
              className="footer"
            >
              <p>
                © 2018. Iveren Shaguy. All images were gotten from Google and are not mine.
              </p>
            </div>
          </Footer>
        </div>
      </CatererView>
      <Connect(Modal)
        submitError={null}
        submitting={false}
      >
        <Modal
          open={false}
          submitError={null}
          submitting={false}
          toggleModal={[Function]}
          type={null}
        />
      </Connect(Modal)>
    </Meals>
  </Connect(Meals)>
</Provider>
`;

exports[`Meals renders correctly when not fetching 1`] = `
<React.Fragment>
  <CatererView
    isFetching={false}
    logout={[MockFunction]}
    showTime={true}
    type="meals"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <React.Fragment>
      <div
        className="content-wrapper meals caterer-meals"
      >
        <div
          className="top"
        >
          <button
            className="btn btn-pri"
            id="add-meal-btn"
            onClick={[Function]}
          >
            Add a New Meal
          </button>
        </div>
        <div
          className="card-group meals-wrapper"
          id="card-group"
        >
          <MealCard
            key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
            meal={
              Object {
                "description": "Sharwama contains no animal products, perfect for dieters",
                "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                "price": "1200.00",
                "title": "Vegetable Sharwama and Guava Smoothie",
                "vegetarian": true,
              }
            }
            toggleModal={[Function]}
            type="caterer"
          />
          <MealCard
            key="81211c24-51c0-46ec-b1e0-18db55880958"
            meal={
              Object {
                "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                "price": "1500.00",
                "title": "Jollof Rice, Beef and Plantain",
                "vegetarian": false,
              }
            }
            toggleModal={[Function]}
            type="caterer"
          />
        </div>
      </div>
    </React.Fragment>
  </CatererView>
  <Connect(Modal)
    submitError={null}
    submitting={false}
  />
</React.Fragment>
`;

exports[`Meals renders message when not fetching and there are no meals 1`] = `
<React.Fragment>
  <CatererView
    isFetching={true}
    logout={[MockFunction]}
    showTime={true}
    type="meals"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <React.Fragment>
      <div
        className="content-wrapper meals caterer-meals"
      >
        <div
          className="top"
        >
          <button
            className="btn btn-pri"
            id="add-meal-btn"
            onClick={[Function]}
          >
            Add a New Meal
          </button>
        </div>
        <p
          className="text-center"
        >
          You Do Not Have Any Meals Yet
        </p>
      </div>
    </React.Fragment>
  </CatererView>
  <Connect(Modal)
    submitError={null}
    submitting={false}
  />
</React.Fragment>
`;
